src/pybind/mgr/mirroring/fs/snapshot_mirror.py

```diff
@@ -722,6 +722,20 @@ def remove_dir(self, filesystem, dir_path):
         except Exception as e:
             return e.args[0], '', 'failed to remove directory'
 
+    def list_dirs(self, filesystem):
+        try:
+            with self.lock:
+                if not self.filesystem_exist(filesystem):
+                    raise MirrorException(-errno.ENOENT, f'filesystem {filesystem} does not exist')
+                fspolicy = self.pool_policy.get(filesystem, None)
+                if not fspolicy:
+                    raise MirrorException(-errno.EINVAL, f'filesystem {filesystem} is not mirrored')
+                return 0, json.dumps(list(fspolicy.policy.dir_states.keys())), ''
```

Comment:
nit; iirc, we indent this by 4 (spaces).

Reply:
It is 4 spaces already?

Reply:
oh, I mean the JSON ;P